Sage/carbon
# refactor(focus-trap): prevent act warnings in tests when inputs are focused FE-5218

fix #5023

### Proposed behaviour


Refactors how the inputs register that they have been focused to `FocusTrap` to prevent `act`
warnings around state updates in unit tests

### Current behaviour


Inputs register themselves via `focusin` listener which causes `act` warnings exist when testing `FocusTrap`

### Checklist




#### QA


### Additional context



### Testing instructions


- Run `FocusTrap` unit tests, I updated them to remove `act`, there should be no warnings relating to it
- Add my change to `foucs-trap.component` to the branch in the linked issue and run the tests, there should be no warnings
- Test sandbox I've added to ensure that Modal stacking behaves as it does in master

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.


https://codesandbox.io/s/immutable-hill-kfxuu4?file=/src/index.js
